ports/mimxrt/boards/MIMXRT1010_EVK/lpdev_iomux.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Robert Hammelrath
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "fsl_iomuxc.h"
+
+// set the UART IOMUX for MIMXRT1010-EVK
+// Input is the HW UART number
+// LPUART1 on D0/D1 -> 1
+// LPUART3 on A0/D4 -> 3
+// LPUART4 on D6/D7 -> 2
+
+bool lpuart_set_iomux(int8_t uart) {
+
+    switch (uart) {
+        case 1:  // LPUART1
+            IOMUXC_SetPinMux(IOMUXC_GPIO_10_LPUART1_TXD, 0U);
+            IOMUXC_SetPinConfig(IOMUXC_GPIO_10_LPUART1_TXD, 0x10B0u);
+
+            IOMUXC_SetPinMux(IOMUXC_GPIO_09_LPUART1_RXD, 0U);
+            IOMUXC_SetPinConfig(IOMUXC_GPIO_09_LPUART1_RXD, 0x10B0u);
```

Comment:
Is it possible to reduce all this code duplication (for each board) by providing a table of pins, eg:
```c
pins = {
    {IOMUXC_GPIO_10_LPUART1_TXD, 0U, IOMUXC_GPIO_10_LPUART1_TXD, 0x10B0u, ...}, // LPUART1
    {0, 0, 0, 0, 0, 0, 0, 0}, // LPUART2 not supported
    {....}, // LPUART3
};
```

Reply:
I had an initial version which was based on tables. That was not efficient and hard to maintain. Your suggestion looks more compact, but it still has to be MCU- and board-specific, because the MCUs have the pins configured at different pins, and the boards have different UARTs exposed to their pins. Also, these IOMUXC_GPIO_10_LPUART1_TXD macros are effectively sets of five values, which have to be expanded when using them.
Besides that, alphafred and me are considering to replace the existing Pin defintion scheme with pins.csv and MIMXRT10xx-af.csv by a method, which parses fsl_iomuxc.h (thats where the above macro is from), replacing the MIMXRT10xx-af.csv files. This file contains all information needed. That would then replace these lpdev_iomuxc.c files. But that is still to come. Alphafred is at the moment busy with implementing SD card support.

Reply:
Ok, le'ts keep it as-is for now.

If you can eventually reduce duplicated code among boards, that would be a good goal.

Reply:
I just started to implement your suggestion to see, if there is a benefit in code size.
The duplication among boards affects Teensy 4.0 and Teensy 4.1, which have a similar board set-up, and MIMXRT1050, 1060 and 1064, which have the same Pin setup on the board. These could re-use the same file.

Reply:
So, the implementation increased the Teensy 4.0 code by 52 bytes, and decreased the Teensy 4.1 code bye 8 bytes. But the code looks more clean.

Reply:
Looks much cleaner!